kong/tools/queue.lua

```diff
@@ -506,6 +514,21 @@ local function enqueue(self, entry)
     return nil, "entry must be a non-nil Lua value"
   end
 
+  if self.concurrency_limit == -1 then -- unlimited concurrency
```

Comment:
Usually `0` means unlimited.

Reply:
The negative value is a good candidate of unlimited compared to 0. zero commonly means disabled.